library/network_connections.py

```diff
@@ -1299,7 +1299,14 @@ def check_deactivated_cb():
             finally:
                 ac.handler_disconnect(ac_id)
 
+    def reapply(self, interface, connection):
+        device = self.nmclient.get_device_by_iface(interface)
+        if not device:
+            return False
 
+        version_id = 0
+        flags = 0
+        return Util.call_async(device, "reapply", [connection, version_id, flags])
```

Comment:
I don't think this is right.

For one, the role should not pass a `connection`. Instead, it should first make sure that the connection profile is up to date, and then pass no connection.

Also, the selection of the device doesn't seem right. The interface name does not matter because you can have profile that are not tied to a device by interface-name. What matters is whether the profile you want to activate is already active (on the right device). So, you need to find the profile, then the device(s) on which it's active, and if that the right one, then call reapply.